### PR TITLE
Enhanced chain vars

### PR DESCRIPTION
**Pull request by manasdk**
* Ability to reference parameters of the action chain itself in the vars of an actionchain. This provides us an ability to parse parameters and apply some transformation which can be later used in the actionchain.

### CODE REVIEW

File: st2actions/st2actions/runners/actionchainrunner.py [hunk 1]
```diff
@@ -53,8 +53,12 @@ def __init__(self, chainspec, chainname):
         LOG.debug('Using %s as default for %s.', self.actionchain.default, self.chainname)
         if not self.actionchain.default:
             raise Exception('Failed to find default node in %s.' % (self.chainname))
-        # finalize the vars and save them around to be used at execution time.
-        self.vars = self._get_rendered_vars(self.actionchain.vars) if self.actionchain.vars else {}
+        self.vars = {}
+
+    def init_vars(self, action_parameters):
+        if self.actionchain.vars:
+            self.vars = self._get_rendered_vars(self.actionchain.vars,
```
Comment (lakshmi-kannan) [inline st2actions/st2actions/runners/actionchainrunner.py]: This change will work but I am wondering why we aren't letting this code https://github.com/StackStorm/st2/blob/enhanced_chain_vars/st2actions/st2actions/runners/actionchainrunner.py#L330 render vars before evaluating action_params?

Comment (lakshmi-kannan) [inline st2actions/st2actions/runners/actionchainrunner.py]: Never mind. Then we have to render this for every task in the action.

**Comment by lakshmi-kannan** [comment]
LGTM. Using action_params in vars is a little awkward but sure.

File: st2actions/st2actions/runners/actionchainrunner.py (hunk 1 above)
Comment (manasdk) [inline st2actions/st2actions/runners/actionchainrunner.py]: :)

**Comment by manasdk** [comment]
Ignoring scrutinizer. Cannot work out why that fails.

File: st2actions/st2actions/runners/actionchainrunner.py [hunk 2]
```diff
@@ -53,8 +53,12 @@ def __init__(self, chainspec, chainname):
         LOG.debug('Using %s as default for %s.', self.actionchain.default, self.chainname)
         if not self.actionchain.default:
             raise Exception('Failed to find default node in %s.' % (self.chainname))
-        # finalize the vars and save them around to be used at execution time.
-        self.vars = self._get_rendered_vars(self.actionchain.vars) if self.actionchain.vars else {}
+        self.vars = {}
+
+    def init_vars(self, action_parameters):
+        if self.actionchain.vars:
```
Comment (Kami) [inline st2actions/st2actions/runners/actionchainrunner.py]: What happens if `self.actionchain.vars` is not defined? It looks like we just silently ignore that condition.

Is that desirable / a good idea?

**Comment by Kami** [comment]
We need some docs on the whole context and variables availability - which variables are available in the context where and when.

**Pull request merged**